main.cpp

```diff
@@ -274,8 +220,11 @@ DoubleType& DoubleType::divide(double rhs)
         std::cout << "warning: floating point division by zero!" << std::endl;
 
     *value /= rhs;
-    return *this;
-}// ======================================== Int primitive ==========
+    return *this;        
+}
+
+DoubleType::operator double() const { return *DoubleType::value; }
```

Comment:
This is a member function.
you don't need to fully qualify the member variable in the function body.
- [ ] revise